File: LexTMLParser.cpp
// LexTMLParser.cpp
#include "LexTMLParser.h"
#include <regex>
#include <sstream>

LexTML parseLexTML(const std::string& tml) {
    std::regex pattern("^LX([A-Z]{2})([1-6])([a-z])([1-6])-([a-z]{2}\+[a-z]{2})@([a-zA-Z0-9_]+)((\+[a-z]{2})*)$");
    std::smatch match;
    LexTML result;

    if (std::regex_match(tml, match, pattern)) {
        result.emotion = match[1];
        result.intensity = std::stoi(match[2]);
        result.nuance = match[3].str()[0];
        result.expression = std::stoi(match[4]);
        result.symbolicGesture = match[5];
        result.target = match[6];

        std::string modStr = match[7];
        std::stringstream ss(modStr);
        std::string item;
        while (getline(ss, item, '+')) {
            if (!item.empty()) result.modifiers.push_back(item);
        }
    }

    return result;
}

File: lexTML-parser.js
// LexCode JavaScript Sample Parser
function parseLexCode(code) {
  const pattern = /^LX([A-Z]{2})([1-6])([a-z])([1-6])-([a-z]{2}\+[a-z]{2})@([a-zA-Z0-9_]+)((\+[a-z]{2})*)$/;
  const match = code.match(pattern);
  if (!match) return null;
  
  return {
    emotion: match[1],
    intensity: parseInt(match[2]),
    nuance: match[3],
    expression: parseInt(match[4]),
    symbolicGesture: match[5],
    target: match[6],
    modifiers: match[7] ? match[7].split('+').filter(Boolean) : []
  };
}

// Example
console.log(parseLexCode("LXHC2w3-ri+pb@child+dn+rb")); // Custom robot gesture with rear balance
